test-suite/src/snapshot_integration_tests.rs

```diff
@@ -0,0 +1,72 @@
+use spatialos_sdk::worker::{entity::Entity, snapshot::*, EntityId};
+use std::env;
+use std::f64::EPSILON;
+
+use crate::generated::improbable::*;
+use std::collections::BTreeMap;
+
+#[test]
+pub fn create_and_read_snapshot() {
+    let mut snapshot_path = env::temp_dir();
+    snapshot_path.push("test.snapshot");
```

Comment:
This can be expressed more succinctly as `let snapshot_path = env::temp_dir().join("test.snapshot");`.